### PR TITLE
Remove unnecessary step to setup go environment

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@TylerHelmuth these are from long time ago, before using "setup-go" action when we were using containers with golang pre-installed. Back then we had to configure the environment, after we changed to setup-go no need.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -417,10 +385,8 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version: 1.17
-      - name: Setup Go Environment
+      - name: Mkdir bin and dist
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:388]: setup-go creates `~/go/bin` not `~/bin`

[the remark this answers is not in the document]
Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:388]: the `/bin` and `/dist` are for our builds not for external go tools/binaries

**Review by djaglowski — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Please rebase to fix the failing check

**Pull request merged**